cmake/AddBipedalLocomotionUnitTest.cmake

```diff
@@ -5,6 +5,23 @@
 # This software may be modified and distributed under the terms of the
 # BSD-3-Clause license. See the accompanying LICENSE file for details.
 
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+# Fetch catch2 only if the testing are built
+if (BUILD_TESTING)
+  include(FetchContent)
+  FetchContent_Declare(Catch2
+    GIT_REPOSITORY https://github.com/catchorg/Catch2.git
+    GIT_TAG        v2.13.8)
+
+  FetchContent_GetProperties(Catch2)
+  if(NOT Catch2_POPULATED)
+    message(STATUS "Fetching Catch2...")
+    FetchContent_MakeAvailable(Catch2)
+  endif()
+endif()
```

Comment:
What about the case in which ``Catch2`` is already available in the system?

Reply:
I've tested it in my system. The version installed in the system is not considered.

Reply:
You could add an option to allow using the system one. This can be useful if you want to test something with a different version of Catch. See for example https://github.com/ami-iit/matio-cpp/blob/1b41fe4705b4f3cb63732772ca3d5c408d6fb09b/CMakeLists.txt#L103-L118

Reply:
Done in https://github.com/ami-iit/bipedal-locomotion-framework/commit/9d7582295758ac687fc5e8228526f767d2320b0b